src/test/java/it/reply/orchestrator/service/CloudProviderEndpointServiceTest.java

```diff
@@ -0,0 +1,136 @@
+		Deployment deployment = ControllerTestUtils.createDeployment();
+		RankCloudProvidersMessage rcpm = new RankCloudProvidersMessage();
+
+		
+		assertEquals(cloudProviderEndpointServiceImpl.chooseCloudProvider(deployment, rcpm), null);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void failGetCloudProviderEndpoint(){
+		CloudProvider chosenCloudProvider = new CloudProvider();
+		List<PlacementPolicy> placementPolicies = new ArrayList<>();
+		
+		assertEquals(cloudProviderEndpointServiceImpl.getCloudProviderEndpoint(chosenCloudProvider, placementPolicies), null);		
+	}
+	
+	@Test
+	public void getCloudProviderEndpoint(){
+		
+		// OCCI
+		List<PlacementPolicy> placementPolicies = new ArrayList<>();
+		List<CloudService> cmbdProviderServicesByType = new ArrayList<>();
+				
+		CloudProvider chosenCloudProvider = Mockito.mock(CloudProvider.class);
+		
+		CloudServiceData cloudServiceDataOCC = new CloudServiceData();
+		cloudServiceDataOCC.setServiceType("eu.egi.cloud.vm-management.occi");
+		cloudServiceDataOCC.setEndpoint("www.endpoint.com");
+		
+		CloudService cloudService = new CloudService();
+		cloudService.setId(UUID.randomUUID().toString());
+		cloudService.setData(cloudServiceDataOCC);
+		cmbdProviderServicesByType.add(cloudService);
+		
+		Mockito.when(chosenCloudProvider.getCmbdProviderServicesByType(Type.COMPUTE)).thenReturn(cmbdProviderServicesByType);
+		
+		// Result OCCI
+		CloudProviderEndpoint result = new CloudProviderEndpoint();
+		result.setCpEndpoint(cloudService.getData().getEndpoint());
+		result.setCpComputeServiceId(cloudService.getId());
+		result.setIaasType(IaaSType.OCCI);
+		
+		Assert.assertEquals(cloudProviderEndpointServiceImpl.getCloudProviderEndpoint(chosenCloudProvider, placementPolicies), result);
+
+		// OPENSTACK
+		
+		cloudServiceDataOCC.setServiceType("eu.egi.cloud.vm-management.openstack");
+		cloudService.setData(cloudServiceDataOCC);
+		cmbdProviderServicesByType.clear();
+		cmbdProviderServicesByType.add(cloudService);
+		Mockito.when(chosenCloudProvider.getCmbdProviderServicesByType(Type.COMPUTE)).thenReturn(cmbdProviderServicesByType);
+		
+		//result OPENSTACK
+		result.setIaasType(IaaSType.OPENSTACK);
+		
+		Assert.assertEquals(cloudProviderEndpointServiceImpl.getCloudProviderEndpoint(chosenCloudProvider, placementPolicies), result);
+
+		// OPENNEBULA
+		
+		cloudServiceDataOCC.setServiceType("eu.egi.cloud.vm-management.opennebula");
+		cloudService.setData(cloudServiceDataOCC);
+		cmbdProviderServicesByType.clear();
+		cmbdProviderServicesByType.add(cloudService);
+		Mockito.when(chosenCloudProvider.getCmbdProviderServicesByType(Type.COMPUTE)).thenReturn(cmbdProviderServicesByType);
+		
+		// result OPENEBULA
+		result.setIaasType(IaaSType.OPENNEBULA);
+
+		Assert.assertEquals(cloudProviderEndpointServiceImpl.getCloudProviderEndpoint(chosenCloudProvider, placementPolicies), result);
+		
+		// AWS Not covered because contains a lambda expression

```

Comment:
Maybe Cobertura doesn't show the coverage for it. It doesn't mean though that it should not be tested